pkg/cli/server/server.go

```diff
@@ -273,11 +297,25 @@ func run(app *cli.Context, cfg *cmds.Server) error {
 	agentConfig.DataDir = filepath.Dir(serverConfig.ControlConfig.DataDir)
 	agentConfig.ServerURL = url
 	agentConfig.Token = token
-	agentConfig.DisableLoadBalancer = true
+	agentConfig.DisableLoadBalancer = !serverConfig.DisableServer
 	agentConfig.Rootless = cfg.Rootless
-	if agentConfig.Rootless {
-		// let agent specify Rootless kubelet flags, but not unshare twice
-		agentConfig.RootlessAlreadyUnshared = true
+
+	if serverConfig.DisableServer {
+		agentConfig.LBServerPort = lbServerPort
+		for {
+			serverIP, serverPort, err := etcd.GetServerURLFromETCD(ctx, &serverConfig.ControlConfig)
+			if err != nil {
+				logrus.Error(err)
```

Comment:
Is this the point that might benefit from being a Warn rather than an Error?

Reply:
no not really, will fix